flytekit/clis/sdk_in_container/backfill.py

```diff
@@ -0,0 +1,176 @@
+    type=str,
+    default="flytesnacks",
+    help="Project to register and run this workflow in",
+)
+@click.option(
+    "-d",
+    "--domain",
+    required=False,
+    type=str,
+    default="development",
+    help="Domain to register and run this workflow in",
+)
+@click.option(
+    "-v",
+    "--version",
+    required=False,
+    type=str,
+    default=None,
+    help="Version for the registered workflow. If not specified it is auto-derived using the start and end date",
+)
+@click.option(
+    "-n",
+    "--execution-name",
+    required=False,
+    type=str,
+    default=None,
+    help="Create a named execution for the backfill. This can prevent launching multiple executions.",
+)
+@click.option(
+    "--dry-run",
+    required=False,
+    type=bool,
+    is_flag=True,
+    default=False,
+    show_default=True,
+    help="Just generate the workflow - do not register or execute",
+)
+@click.option(
+    "--parallel/--serial",
+    required=False,
+    type=bool,
+    is_flag=True,
+    default=False,
+    show_default=True,
+    help="All backfill can be run in parallel - with max-parallelism",
+)
+@click.option(
+    "--no-execute",
+    required=False,
+    type=bool,
+    is_flag=True,
+    default=False,
+    show_default=True,
+    help="Generate the workflow and register, do not execute",
+)
+@click.option(
+    "--from-date",
+    required=False,
+    type=DateTimeType(),
+    default=None,
+    help="Date from which the backfill should begin. Start date is inclusive.",
+)
+@click.option(
+    "--to-date",
+    required=False,
+    type=DateTimeType(),
+    default=None,
+    help="Date to which the backfill should run_until. End date is inclusive",
+)
+@click.option(
+    "--backfill-window",
+    required=False,
+    type=DurationParamType(),
+    default=None,
+    help="Timedelta for number of days, minutes hours after the from-date or before the to-date to compute the"
+    " backfills between. This is needed with from-date / to-date. Optional if both from-date and to-date are provided",
+)
+@click.argument(
+    "launchplan",
+    required=True,
+    type=str,
+    # help="Name of launchplan to be backfilled.",
+)
+@click.argument(
+    "launchplan-version",
+    required=False,
+    type=str,
+    default=None,
+    # help="Version of the launchplan to be backfilled, if not specified, the latest version "
+    #      "(by registration time) will be used",
+)
+@click.pass_context
+def backfill(
+    ctx: click.Context,
+    project: str,
+    domain: str,
+    from_date: datetime,
+    to_date: datetime,
+    backfill_window: timedelta,
+    launchplan: str,
+    launchplan_version: str,
+    dry_run: bool,
+    no_execute: bool,
+    parallel: bool,
+    execution_name: str,
+    version: str,
+):
+    from_date, to_date = resolve_backfill_window(from_date, to_date, backfill_window)
+    remote = get_and_save_remote_with_click_context(ctx, project, domain)
+    try:
+        entity = remote.launch_backfill(
```

Comment:
Is it possible that this call does not return an entity? Should we inform the user in that case?

Reply:
it is not, but i cleaned up the comment

Reply:
hmm the if block is just to improve the output for the user right?

Reply:
I was just thinking of the case `entity` being null, but it's alright.